File: 9-Rock-Paper-Scissors/5-improving-rock-paper-scissors.py
from random import randint

player_wins = 0
computer_wins = 0

while player_wins < 2 and computer_wins < 2:

    print(f"Player score: {player_wins} | Computer score: {computer_wins}")
    print('...rock...')
    print('...paper...')
    print('...scissors...')


    player = input("(Enter your choice): ").lower()
    if player == "quit" or player == "q":
        break


    rand_num = randint(0,2)
    if rand_num == 0:
        computer = "rock"
    elif rand_num == 1:
        computer = "paper"
    else:
        computer = "scissors"


    print(f"The computer plays {computer}")


    if player == computer:
        print("It's a tie.")
    elif player == "rock":
        if computer == "scissors":
            print("Player wins!")
            player_wins += 1
        else:
            print("Computer wins :(")
            computer_wins += 1
    elif player == "paper":
        if computer == "rock":
            print("Player wins!")
            player_wins += 1
        else:
            print("Computer wins :(")
            computer_wins += 1
    elif player == "scissors":
        if computer == "paper":
            print("Player wins!")
            player_wins += 1
        else:
            print("Computer wins :(")
            computer_wins += 1
    else:
        print("Something went wrong")

if player_wins > computer_wins:
    print("CONGRATS, YOU WON!")
elif player_wins == computer_wins:
    print("IT'S A TIE")
else:
    print("OH NO :( THE COMPUTER WON...")

    

File: 9-Rock-Paper-Scissors/5-improving-rock-paper-scissors.js
// using npm library prompt-sync
const prompt = require('prompt-sync')()

// custom choice function
function choice(arr) {
	return arr[Math.floor(Math.random() * arr.length)]
}

let playerWins = 0
let computerWins = 0
const winningScore = 3

while (playerWins < winningScore && computerWins < winningScore) {
    console.log(`Player score: ${playerWins} | Computer score: ${computerWins}`)
    console.log('...rock...')
    console.log('...paper...')
    console.log('...scissors...')


    let player = prompt("(Enter your choice): ").toLowerCase()
    if (player === 'quit' || player === 'q') { break }
    
    let computer = choice(['rock', 'paper', 'scissors'])
    console.log(`The computer plays ${computer}`)

    let message = ""

    if (player === computer) {
        message = "It's a tie!"
    }
    else if (player === 'rock') {
        if (computer === 'scissors') {
            message = 'Player wins!'
            playerWins += 1
        } else {
            message = 'Computer wins :('
            computerWins += 1
        }
    }
    else if (player === 'paper') {
        if (computer === 'rock') {
            message = 'Player wins!'
            playerWins += 1
        } else {
            message = 'Computer wins :('
            computerWins += 1
        }
    }
    else if (player === 'scissors') {
        if (computer === 'paper') {
            message = 'Player wins!'
            playerWins += 1
        } else {
            message = 'Computer wins :('
            computerWins += 1
        }
    }
    else {
        message = 'Please enter a valid move'
    }
    console.log(message)
}

if (playerWins > computerWins) {
    console.log("CONGRATS, YOU WON!")
} else if (playerWins === computerWins) {
    console.log("IT'S A TIE")
} else {
    console.log("OH NO :( THE COMPUTER WON...")
}